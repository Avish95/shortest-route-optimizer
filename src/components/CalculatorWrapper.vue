<template>
    <div class="loadingWraper" v-if="loading">
        <IconLoading style="width: 5rem;" />
        <span class="loadingText">Calculating the Shortest Path...</span>
    </div>
    <div v-else class="calc_wrapper">
        <div class="">
            <CalculatorInputs @action="handleAction" @clear="result = null" />
        </div>

        <CalcResult v-if="result !== null" :result="result" />
        <div v-else class="image_wrapper">
            <img :src="image" alt="graphical image of a sitting man">
        </div>
    </div>


</template>
<script setup lang="ts">
import image from '@/assets/calcImage.png'
import CalculatorInputs from './CalcInputs.vue'
import CalcResult from './CalcResult.vue'
import type Result from '@/types/result'
import IconLoading from './icons/IconLoading.vue'
import { ShortestPath } from '@/services/calculator'
import { ref } from 'vue'

const handleAction = async (event) => {
    loading.value = true
    ShortestPath(event).then((res) => {
        result.value = res

    }).finally(() => {
        loading.value = false
    })

}
const loading = ref(false)
const result = ref<Result | null>(null)
</script>
<style scoped>
.calc_wrapper {
    background-color: white;
    border-radius: 0.5rem;
    display: grid;
    grid-template-columns: 1fr 1fr;
    overflow: hidden;
}

.image_wrapper {
    display: flex;
    justify-content: center;
    align-items: center;
}

.loadingWraper {
    display: flex;
    flex-direction: column;
    gap: 1.5rem;
    justify-content: center;
    align-items: center;
}

.loadingText {
    color: var(--primary-200);
    font-family: 'Poppins';
    font-size: 2rem;
    text-align: center;
}

@media (max-width: 900px) {
    .calc_wrapper {
        display: flex;
        flex-direction: column;
    }

    
}
</style>