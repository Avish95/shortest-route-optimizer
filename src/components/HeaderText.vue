<template>
    <div class="heading_container">
        <h1>Dijiktraâ€™s Algorithm Calculator</h1>
        <p class="hero_text">Discovering Optimal Routes Through Nodes Using Dijkstra's Method</p>
    </div>
</template>

<style scoped>
.heading_container {
    width: 100%;
    display: flex;
    flex-direction: column;
    gap: 1rem;
    justify-content: center;
}

.hero_text {
    color: white;
    text-align: center;
    font-size: 1.125rem;
    line-height: 1.5rem;
    font-weight: 400;
}

@media (max-width: 1200px) {
    .heading_container {
        width: 100vw;
        padding: 0 1rem;
        gap: 0
    }

    .hero_text {
        font-size: 0.5rem;
        line-height: 1.5rem;
        text-align: left;
    }
}
</style>