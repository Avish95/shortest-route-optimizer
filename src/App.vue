<script setup lang="ts">
import HeaderText from './components/HeaderText.vue'
import CalculatorWrapper from './components/CalculatorWrapper.vue'
</script>

<template>
  <header class="header">
    <HeaderText />
  </header>
  <main class="main-centered">
    <CalculatorWrapper />
  </main>
</template>

<style scoped>
.main-centered {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  position: relative;

}

.header {
  position: absolute;
  top: 4rem;
  left: 50%;
  transform: translateX(-50%);
  z-index: 10;
}

@media (max-width: 1200px) {
  .header {
    top: 3rem;
  }
  .main-centered {
    padding: 0.75rem;
  }
}

/* Media Query for Screens Up to 900px */
@media (max-width: 900px) {
  .header {
    top: 2rem;
  }

  main {
    width: 100%;
  }
}
</style>
