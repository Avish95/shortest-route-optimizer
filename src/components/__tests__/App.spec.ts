/* eslint-disable @typescript-eslint/ban-ts-comment */
// @ts-nocheck
import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import App from '@/App.vue'
import HeaderText from '@/components/HeaderText.vue'
import CalculatorWrapper from '@/components/CalculatorWrapper.vue'

describe('App', () => {
  it('renders properly', () => {
    const wrapper = mount(App)
    expect(wrapper.exists()).toBe(true)
  })

  it('renders HeaderText component', () => {
    const wrapper = mount(App)
    const headerText = wrapper.findComponent(HeaderText)
    expect(headerText.exists()).toBe(true)
  })

  it('renders CalculatorWrapper component', () => {
    const wrapper = mount(App)
    const calculatorWrapper = wrapper.findComponent(CalculatorWrapper)
    expect(calculatorWrapper.exists()).toBe(true)
  })

  it('matches the snapshot', () => {
    const wrapper = mount(App)
    expect(wrapper.html()).toMatchSnapshot()
  })

 
})
