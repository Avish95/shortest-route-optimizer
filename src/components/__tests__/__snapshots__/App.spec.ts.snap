// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches the snapshot 1`] = `
"<header data-v-7a7a37b1="" class="header">
  <div data-v-961aa8b9="" data-v-7a7a37b1="" class="heading_container">
    <h1 data-v-961aa8b9="">Dijiktraâ€™s Algorithm Calculator</h1>
    <p data-v-961aa8b9="" class="hero_text">Discovering Optimal Routes Through Nodes Using Dijkstra's Method</p>
  </div>
</header>
<main data-v-7a7a37b1="" class="main-centered">
  <div data-v-d9381930="" data-v-7a7a37b1="" class="calc_wrapper">
    <div data-v-d9381930="" class="">
      <div data-v-f1588342="" data-v-d9381930="" class="Input_wrapper">
        <h2 data-v-f1588342="">Select Path</h2>
        <div data-v-204186c5="" data-v-f1588342="" class="select_wrapper"><label data-v-204186c5="" for="select_from">From Node</label><select data-v-204186c5="" name="select_from" id="select_from" class="">
            <option data-v-204186c5="" disabled="" selected="" hidden="" value="">Select</option>
            <option data-v-204186c5="" value="A">A</option>
            <option data-v-204186c5="" value="B">B</option>
            <option data-v-204186c5="" value="C">C</option>
            <option data-v-204186c5="" value="D">D</option>
            <option data-v-204186c5="" value="E">E</option>
            <option data-v-204186c5="" value="F">F</option>
            <option data-v-204186c5="" value="G">G</option>
            <option data-v-204186c5="" value="H">H</option>
            <option data-v-204186c5="" value="I">I</option>
          </select>
          <!--v-if-->
        </div>
        <div data-v-204186c5="" data-v-f1588342="" class="select_wrapper"><label data-v-204186c5="" for="select_from">To Node</label><select data-v-204186c5="" name="select_from" id="select_from" class="">
            <option data-v-204186c5="" disabled="" selected="" hidden="" value="">Select</option>
            <option data-v-204186c5="" value="A">A</option>
            <option data-v-204186c5="" value="B">B</option>
            <option data-v-204186c5="" value="C">C</option>
            <option data-v-204186c5="" value="D">D</option>
            <option data-v-204186c5="" value="E">E</option>
            <option data-v-204186c5="" value="F">F</option>
            <option data-v-204186c5="" value="G">G</option>
            <option data-v-204186c5="" value="H">H</option>
            <option data-v-204186c5="" value="I">I</option>
          </select>
          <!--v-if-->
        </div>
        <div data-v-f1588342="" class="buttons_wrapper"><button data-v-f1588342="" class="clear_btn">Clear</button><button data-v-f1588342="" class="primary_btn"> Calculate <i data-v-f1588342=""><svg data-v-f1588342="" width="14" height="18" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path d="M2 17.333h10a1.666 1.666 0 0 0 1.667-1.666V2.333A1.667 1.667 0 0 0 12 .667H2A1.667 1.667 0 0 0 .333 2.333v13.334A1.666 1.666 0 0 0 2 17.333Zm2.5-2.5H2.833v-1.666H4.5v1.666Zm0-3.333H2.833V9.833H4.5V11.5Zm0-3.333H2.833V6.5H4.5v1.667Zm3.333 6.666H6.167v-1.666h1.666v1.666Zm0-3.333H6.167V9.833h1.666V11.5Zm0-3.333H6.167V6.5h1.666v1.667Zm3.334 6.666H9.5v-5h1.667v5Zm0-6.666H9.5V6.5h1.667v1.667ZM2 2.333h10v2.5H2v-2.5Z" fill="#fff"></path>
              </svg></i></button></div>
      </div>
    </div>
    <div data-v-d9381930="" class="image_wrapper"><img data-v-d9381930="" src="/src/assets/calcImage.png" alt="graphical image of a sitting man"></div>
  </div>
</main>"
`;
